src/utils/polyfillEnvironment.js

```diff
@@ -42,7 +42,9 @@ const scriptURL = require('react-native').NativeModules.SourceCode.scriptURL; //
 // or Android device it will be `localhost:<port>` but when using real iOS device
 // it will be `<ip>.xip.io:<port>`. Thus the code below ensure we connect and download
 // manifest/hot-update from a valid origin.
-global.DEV_SERVER_ORIGIN = scriptURL.match(/(^.+)\/index/)[1];
+const match = scriptURL.match(/(^.+)\/index/);
```

Comment:
Let's do sanity check here

`const match = scriptURL && scriptURL.match(/(^.+)\/index/);`

Reply:
Based on @czystyl findings `scriptURL` is always set, it's just different for development and production envs

Reply:
Would it be a problem to do that sanity check tho?

Reply:
That's true

Reply:
I added this check if RN change something.